graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1579,25 +1584,55 @@ def write_edge_structure(
         edge_df_with_int_ids = edge_df_with_int_ids.drop(src_col, dst_col).repartition(
             self.num_output_files
         )
-        edge_df_with_int_ids_and_all_features = edge_df_with_int_ids
-        edge_df_with_only_int_ids = edge_df_with_int_ids.select(["src_int_id", "dst_int_id"])
 
         edge_structure_path = os.path.join(
             self.output_prefix, f"edges/{edge_type.replace(':', '_')}"
         )
-        logging.info("Writing edge structure for edge type %s...", edge_type)
-        if self.add_reverse_edges:
+
+        edge_df_with_int_ids_and_all_features = edge_df_with_int_ids
+        edge_df_with_only_int_ids = edge_df_with_int_ids.select(["src_int_id", "dst_int_id"])
+        if self.add_reverse_edges and not self.is_homogeneous:
             edge_df_with_only_int_ids.cache()
-        path_list = self._write_df(edge_df_with_only_int_ids, edge_structure_path)
 
-        if self.add_reverse_edges:
+        if self.add_reverse_edges and not self.is_homogeneous:
+            logging.info("Writing edge structure for edge type %s...", edge_type)
+            path_list = self._write_df(edge_df_with_only_int_ids, edge_structure_path)
             reversed_edges = edge_df_with_only_int_ids.select("dst_int_id", "src_int_id")
             reversed_edge_structure_path = os.path.join(
                 self.output_prefix, f"edges/{rev_edge_type.replace(':', '_')}"
             )
             logging.info("Writing edge structure for reverse edge type %s...", rev_edge_type)
             reverse_path_list = self._write_df(reversed_edges, reversed_edge_structure_path)
+        elif self.add_reverse_edges and self.is_homogeneous:
+            # Homogeneous graph will only add reverse edges to the same edge type
+            # instead of creating a {relation_type}-rev edge type.
+            other_columns = [
+                c
+                for c in edge_df_with_int_ids_and_all_features.columns
+                if c not in ("src_int_id", "dst_int_id")
+            ]
+            reversed_edges = edge_df_with_int_ids_and_all_features.select(
+                col("dst_int_id").alias("src_int_id"),
+                col("src_int_id").alias("dst_int_id"),
+                *other_columns,
+            ).withColumn(HOMOGENEOUS_REVERSE_COLUMN_FLAG, F.lit(False))
+            edge_df_with_int_ids_and_all_features = (
+                edge_df_with_int_ids_and_all_features.withColumn(
+                    HOMOGENEOUS_REVERSE_COLUMN_FLAG, F.lit(True)
```

Comment:
Is HOMOGENEOUS_REVERSE_COLUMN_FLAG used somewhere downstream?

Reply:
HOMOGENEOUS_REVERSE_COLUMN_FLAG is to identify if the edge is reversed edge, if it is, then do not assign mask to the edge.